tests/e2e_test.go

```diff
@@ -84,7 +105,8 @@ func TestE2E(t *testing.T) {
 // registered on the P-Chain.
 // Adds two disjoint sets of 5 of the new validator nodes to validate two new subnets with a
 // a single Subnet-EVM blockchain.
-var _ = ginkgo.BeforeSuite(func() {
+var _ = ginkgo.SynchronizedBeforeSuite(func() []byte {
```

Comment:
`SynchronizedBeforeSuite` takes two functions as parameters. The first function is run on a single core, while all other cores wait. The first function returns `[]byte` which are passed to the second function. The second function runs in parallel on all cores. Individual suites (ginkgo `Describe` blocks) are distributed among the parallel runners.

To accomodate this framework we have to:
- Define global variables at file scope. Each parallel runner _captures its own copy_ of these global variables, so we have to set them in the second function of `SynchronizedBeforeSuite`. If they are used in the first function, they need to be set there are well.
- Give each parallel runner a unique EVM address, otherwise we'll get nonce conflicts. Tests don't necessarily need to have unique addresses, just as long as the addresses are not used simultaneously. I chose to do this by modifying the genesis file.

Reply:
I think instead of modifying the genesis file, we should keep one funded address/key pair funded in the genesis file, and use a global helper function `FundAddress()`, that takes in an address, and amount, and a slice of WSClients as parameters. It then uses the global funded address to transfer funds to whatever wallet the test needs. This way we can dynamically allocate the funded address for most tests, and can also statically allocate one if we want. You'd have to put a mutex on `FundAddress()`, but I don't think it would add much overhead, and we gain scaleability.